dist/tools/compile_test/compile_test.py

```diff
@@ -100,16 +100,21 @@ def build_all():
             stdout.write('\tBuilding application: {} ({}/{}) '.format(colorize_str(application, Termcolor.blue), nth, len(applications)))
             stdout.flush()
             try:
+                app_dir = join(riotbase, folder, application)
                 subprocess = Popen(('make', 'buildtest'),
                                    bufsize=1, stdin=null, stdout=PIPE, stderr=null,
-                                   cwd=join(riotbase, folder, application),
+                                   cwd=app_dir,
                                    env=subprocess_env)
 
                 results, results_with_output = tee(get_results_and_output_from(subprocess.stdout))
                 results = groupby(sorted(results), lambda res: res[0])
                 results_with_output = list(filter(lambda res: res[2].getvalue(), results_with_output))
                 failed_with_output = list(filter(lambda res: 'failed' in res[0], results_with_output))
                 success_with_output = list(filter(lambda res: 'success' in res[0], results_with_output))
+                # check if bin-directory isn't in system's PATH to not accidentally
+                # delete some valuable system executable ;-)
+                if join(app_dir, "bin") not in environ.get("PATH", "/bin:/usr/bin:/usr/local/bin:"):
```

Comment:
I think there's missing a .split(":") if I am not mistaken. Also, it might be just as well to let the default value be "" and always append the /bin /usr/bin /usr/local/bin paths regardless of whether they are in $PATH or not.

Reply:
Nope its a string and `in` for strings is "string contains other string" so a split would just result in more operations than necessary.